### PR TITLE
[Skeleton Tool] Automatic change Bone Width parameter in Tool Options Panel (for Skeleton Deformation)

### LINKED ISSUE #1658: [Skeleton Tool] Automatic change Bone Width parameter in Tool Options Panel (for Skeleton Deformation)
When working with Skeleton Tool and Skeleton Deformation Layer:
If user changes width of bone tip, then the new value of width should be applied to Bone Width parameter in Tool Options Panel

Explanation video - https://youtu.be/SyZkaGEe5zI

### PR DESCRIPTION
**Pull request by Aa20475**
Closes #1658

### CODE REVIEW

**Comment by morevnaproject** [comment]
Hi @Aa20475! Can you please cleanup this PR? (I see that only last commit should be in this PR...) ^__^

**Comment by Aa20475** [comment]
Yes sure! I'll do it asap.. my exams are bout to end.. so I'll start working again.

**Comment by morevnaproject** [comment]
Awesome!

**Comment by morevnaproject** [comment]
Hi @Aa20475! Any progress on this? ^__^

**Comment by Aa20475** [comment]
Hey! Please check!

**Comment by morevnaproject** [comment]
@Aa20475 I have verified and it is not on the latest commit. Can you please rebase on top of the latest commit from master?

**Comment by morevnaproject** [comment]
Oh, wait, I can squash this PR. Thank you! ^__^

File: synfig-studio/src/synfigapp/actions/valuedescset.cpp [hunk 1]
```diff
@@ -199,6 +199,64 @@ Action::ValueDescSet::prepare()
...
+					if(!action->is_ready())
+						throw Error(Error::TYPE_NOTREADY);
+
+					add_action(action);
+				}
+			}
+		}
+	}
+
+
```
Comment (rodolforg) [inline synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: @morevnaproject @Aa20475 

Does this stuff do anything? Could you please show me how/what I can make on Synfig to reach this code?
I deleted it , and the (closed) issue doesn't seem to be affected.

Comment (Aa20475) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: As the comment on the top says, when we are using the skeleton tool to edit a `ValueNode_Composite`, the offset between the bones should be maintained.

Comment (rodolforg) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: Whoa! Thank you for quick answer!!

By what do you mean? With or without this code, I move/rotate a bone with Bone Tool and its children moves together with it.

Comment (morevnaproject) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: > its children moves together with it.

@rodolforg This is not for children, it's for second bone from pair (in Skeleton Deformation layer). This code is fixing this issue - https://github.com/synfig/synfig/issues/1627
Without this code we will have wrong behavior, which is demonstrated in this video - https://www.youtube.com/watch?v=dzTPdNbqS3U&feature=youtu.be

Comment (rodolforg) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: It's supposed to work this new fixed way only with Bone Tool?

Comment (morevnaproject) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: Yes, it is supposed to work only when BoneTool is active. On the other hand... your question made me think... Maybe it worth to apply the same behavior for any other tool? How about to try this in separate PR? ^__^

Comment (rodolforg) [reply synfig-studio/src/synfigapp/actions/valuedescset.cpp:259]: You are a synfig user and animator waaaaay more than me XD It's up to you.

**Pull request merged**